docs/user/cli/console.rst

```diff
@@ -0,0 +1,99 @@
+
+        .. code-block:: scala
+
+            scala> val dsParams = Map("fs.path" -> "file:///tmp/fsds/", "fs.encoding" -> "parquet")
+            dsParams: scala.collection.immutable.Map[String,String] = Map(fs.path -> file:///tmp/fsds/, fs.encoding -> parquet)
+
+            scala> val ds = DataStoreFinder.getDataStore(dsParams)
+            ds: org.geotools.data.DataStore = org.locationtech.geomesa.fs.FileSystemDataStore@27a7ef08
+
+
+        Now we do some example discovery to see what feature types and schemas are stored
+        in the database.
+
+        .. code-block:: scala
+
+            scala> ds.getTypeNames()
+            res0: Array[String] = Array(gdelt-quickstart)
+
+            scala> val sft = ds.getSchema("gdelt-quickstart")
+            sft: org.opengis.feature.simple.SimpleFeatureType = SimpleFeatureTypeImpl gdelt-quickstart identified extends Feature(GLOBALEVENTID:GLOBALEVENTID,Actor1Name:Actor1Name,Actor1CountryCode:Actor1CountryCode,Actor2Name:Actor2Name,Actor2CountryCode:Actor2CountryCode,EventCode:EventCode,NumMentions:NumMentions,NumSources:NumSources,NumArticles:NumArticles,ActionGeo_Type:ActionGeo_Type,ActionGeo_FullName:ActionGeo_FullName,ActionGeo_CountryCode:ActionGeo_CountryCode,dtg:dtg,geom:geom)
+
+
+        In order to sample the data we create a ``Query`` and a ``FeatureReader`` and
+        then run the query.
+
+        .. code-block:: scala
+
+            scala> val query = new Query(sft.getName.toString())
+            query: org.geotools.data.Query =
+            Query:
+               feature type: gdelt-quickstart
+               filter: Filter.INCLUDE
+               [properties:  ALL ]
+
+            scala> val reader = ds.getFeatureReader(query, Transaction.AUTO_COMMIT)
+            reader: org.geotools.data.FeatureReader[org.opengis.feature.simple.SimpleFeatureType,org.opengis.feature.simple.SimpleFeature] = org.geotools.data.simple.DelegateSimpleFeatureReader@7bd96822
+
+
+        Finally, we consume the ``FeatureReader``, printing out the results.
+
+        .. code-block:: scala
+
+            scala> while (reader.hasNext()) { println(reader.next().toString()) }
+            ScalaSimpleFeature:719024956:719024956|||GANG||120|6|1|6|1|Brazil|BR|Sun Dec 31 19:00:00 EST 2017|POINT (-55 -10)
+            ScalaSimpleFeature:719024898:719024898|||SYDNEY|AUS|010|14|2|14|4|Sydney, New South Wales, Australia|AS|Sun Dec 31 19:00:00 EST 2017|POINT (151.217 -33.8833)
+            ScalaSimpleFeature:719024882:719024882|SECURITY COUNCIL||PYONGYANG|PRK|163|2|1|2|1|Russia|RS|Sun Dec 24 19:00:00 EST 2017|POINT (100 60)
+            ScalaSimpleFeature:719024881:719024881|||RUSSIA|RUS|042|2|1|2|3|Allegheny County, Pennsylvania, United States|US|Sun Dec 24 19:00:00 EST 2017|POINT (-80.1251 40.6253)
+            ScalaSimpleFeature:719025149:719025149|ARGENTINE|ARG|DIOCESE||010|1|1|1|4|Corrientes, Corrientes, Argentina|AR|Sun Dec 31 19:00:00 EST 2017|POINT (-58.8341 -27.4806)
+            ...
+
```

Comment:
could you add a close on the reader and a dispose on the datastore? while probably not strictly necessary, it's nice to encourage good habits :)

Reply:
Sure